modules/fedimint-wallet/src/lib.rs

```diff
@@ -1351,20 +1417,52 @@ impl Wallet {
     async fn create_peg_out_tx(
         &self,
         dbtx: &mut ModuleDatabaseTransaction<'_, ModuleInstanceId>,
-        peg_out: &PegOut,
+        output: &WalletOutput,
     ) -> Result<UnsignedTransaction, WalletError> {
         let change_tweak = self
             .current_round_consensus(dbtx)
             .await
             .unwrap()
             .randomness_beacon;
-        self.offline_wallet().create_tx(
-            peg_out.amount,
-            peg_out.recipient.script_pubkey(),
-            self.available_utxos(dbtx).await,
-            peg_out.fees.fee_rate,
-            &change_tweak,
-        )
+
+        let mut available_utxos = self.available_utxos(dbtx).await;
+        available_utxos.sort_by_key(|(_, utxo)| utxo.amount);
+
+        match output {
+            WalletOutput::PegOut(peg_out) => self.offline_wallet().create_tx(
+                peg_out.amount,
+                peg_out.recipient.script_pubkey(),
+                available_utxos,
+                peg_out.fees.fee_rate,
+                &change_tweak,
+                None,
+            ),
+            WalletOutput::Rbf(rbf) => {
+                let mut tx = dbtx
+                    .get_value(&PendingTransactionKey(rbf.txid))
+                    .await
+                    .expect("DB error")
+                    .ok_or(WalletError::RbfTransactionIdNotFound)?;
+
+                if tx.fees.total_weight != rbf.fees.total_weight {
+                    return Err(WalletError::RbfTxWeightIncorrect);
+                }
+
+                // Add the rbf fees to the existing tx fees
+                tx.fees.fee_rate.sats_per_kvb += rbf.fees.fee_rate.sats_per_kvb;
+                // Prioritize the existing UXTOs first so we properly RBF
+                tx.selected_utxos.extend(available_utxos);
+
+                self.offline_wallet().create_tx(
+                    tx.peg_out_amount,
+                    tx.destination,
+                    tx.selected_utxos,
+                    tx.fees.fee_rate,
+                    &change_tweak,
+                    Some(rbf.clone()),
+                )
```

Comment:
Afaik there isn't any interface contract in `create_tx` that says the first UTXOs will always be chosen. Maybe we should split the argument into "for sure" inputs and "potential additional inputs"? There should also be a comment about the dangers of not using the previous UTXOs (double spends, Fedimint forgetting about them, …) so someone doesn't "optimize" it later.

On another note, the ordering by amount should probably happen inside `create_tx` since the algorithm in there likely assumes that to be the case.

Reply:
Sure, did think about doing it that way, just thought this was a simpler interface, but does require the caller to order the UTXOs themselves.